airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,571 @@
+        :param dictionary_to_validate: dictionary where the field should be present
+        :type dictionary_to_validate: dict
+        :param parent: full path of parent field
+        :type parent: str
+        :param force_optional: forces the field to be optional
+        (all union fields have force_optional set to True)
+        :type force_optional: bool
+        :return: True if the field is present
+        """
+        field_name = validation_spec['name']
+        field_type = validation_spec.get('type')
+        optional = validation_spec.get('optional')
+        regexp = validation_spec.get('regexp')
+        children_validation_specs = validation_spec.get('fields')
+        required_api = validation_spec.get('api')
+        custom_validation = validation_spec.get('custom_validation')
+
+        full_field_path = self._get_field_name_with_parent(field_name=field_name,
+                                                           parent=parent)
+        if required_api and required_api != self._api:
+            self.log.debug(
+                "Not validating the field '{}' for api version '{}' "
+                "as it is only required for api '{}'".
+                format(field_name, self._api, required_api))
+            return False
+        value = dictionary_to_validate.get(field_name)
+
+        if (optional or force_optional) and value is None:
+            self.log.debug("The optional field '{}' is missing, but that's perfectly ok".
+                           format(full_field_path))
+            return False
+
+        # Certainly down from here the field is present (value is not None)
+        # so we should only return True from now on
+
+        self._sanity_checks(children_validation_specs=children_validation_specs,
+                            field_type=field_type,
+                            full_field_path=full_field_path,
+                            regexp=regexp,
+                            custom_validation=custom_validation,
+                            value=value)
+
+        if regexp:
+            self._validate_regexp(full_field_path, regexp, value)
+        elif field_type == 'dict':
+            if not isinstance(value, dict):
+                raise FieldValidationException(
+                    "The field '{}' should be a dictionary and is '{}'".
+                    format(full_field_path, value))
+            if children_validation_specs is None:
+                self.log.debug(
+                    "The dict field '{}' has no children fields defined, "
+                    "but that's perfectly ok. "
+                    "We just skip validating dict elements.".format(full_field_path))
+            else:
+                self._validate_dict(children_validation_specs, full_field_path, value)
+        elif field_type == 'union':
+            if not children_validation_specs:
+                raise ValidationSpecificationException(
+                    "The union field '{}' has no children "
```

Comment:
Generally, please review all error messages. What are 'children' of the union? Why is it 'clearly' wrong?

How about something like:

Error: no value has been specified for the union field X. Please specify one.

Reply:
Fixed in 3722a2be480c6d8080fc60491e8fa431b7eb6590